location/utils.py

```diff
@@ -54,3 +54,48 @@ def approx_home_location(data, sr_col='stay_region'):
     home = Counter(trg_data[sr_col]).most_common()[0][0]
 
     return home
+
+
+def compute_total_gyration(data, sr_col='stay_region'):
+    """
+    Computes the total radius of gyration.
+
+    Parameters:
+    -----------
+
+    data: DataFrame
+        Location data.
+
+    sr_col: str
+        Column name for stay region.
+        Default is 'stay_region'.
+
+    Returns:
+    --------
+
+    (radius of gyration): float
+        Radius gyration in meter.
+    """
+    loc_data = data.copy()
+    loc_data = loc_data[pd.notnull(loc_data[sr_col])]
+    # different visited locations
```

Comment:
I see, you are using these to count the frequency later. We could have potentially just add each instance instead of weighting:

```
for _, r in loc_data.iterrow():
     p = r.latitude, r.longitude
     d = vincenty(p, r_cm).m
     temp_sum += d ** 2
```

The advantage here is less amount of code and simpler implementation. But, your code might be slightly faster. So, your call.